src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/components/Withdraw.tsx

```diff
@@ -0,0 +1,208 @@
+  )
+
+  const cryptoStakeBalance = useAppSelector(state =>
+    selectDelegationCryptoAmountByAssetIdAndValidator(state, {
+      accountSpecifier,
+      validatorAddress: contractAddress,
+      assetId,
+    }),
+  )
+  const cryptoStakeBalanceHuman = bnOrZero(cryptoStakeBalance).div(`1e+${asset?.precision}`)
+
+  const fiatStakeAmountHuman = cryptoStakeBalanceHuman.times(bnOrZero(marketData.price)).toString()
+
+  if (!state || !dispatch) return null
+
+  const getWithdrawGasEstimate = async () => {
+    if (!state.userAddress) return
+
+    const { gasLimit, gasPrice } = await getStakingFees(asset, marketData.price)
+
+    try {
+      return bnOrZero(gasPrice).times(gasLimit).toFixed(0)
+    } catch (error) {
+      moduleLogger.error(
+        { fn: 'getWithdrawGasEstimate', error },
+        'Error getting deposit gas estimate',
+      )
+      const fundsError =
+        error instanceof Error && error.message.includes('Not enough funds in reserve')
+      toast({
+        position: 'top-right',
+        description: fundsError
+          ? translate('defi.notEnoughFundsInReserve')
+          : translate('common.somethingWentWrong'),
+        title: translate('common.somethingWentWrong'),
+        status: 'error',
+      })
+    }
+  }
+
+  const handleContinue = async (formValues: CosmosWithdrawValues) => {
+    if (!state.userAddress) return
+    // set withdraw state for future use
+    dispatch({
+      type: CosmosWithdrawActionType.SET_WITHDRAW,
+      payload: formValues,
+    })
+    dispatch({
+      type: CosmosWithdrawActionType.SET_LOADING,
+      payload: true,
+    })
+    try {
+      const estimatedGasCrypto = await getWithdrawGasEstimate()
+
+      if (!estimatedGasCrypto) return
+      dispatch({
+        type: CosmosWithdrawActionType.SET_WITHDRAW,
+        payload: { estimatedGasCrypto },
+      })
+      onNext(DefiStep.Confirm)
+      dispatch({
+        type: CosmosWithdrawActionType.SET_LOADING,
+        payload: false,
+      })
+    } catch (error) {
+      moduleLogger.error({ fn: 'handleContinue', error }, 'Error with withdraw')
+      dispatch({
+        type: CosmosWithdrawActionType.SET_LOADING,
+        payload: false,
+      })
+      toast({
+        position: 'top-right',
+        description: translate('common.somethingWentWrongBody'),
+        title: translate('common.somethingWentWrong'),
+        status: 'error',
+      })
+    }
+  }
+
+  const handleCancel = () => {
+    browserHistory.goBack()
+  }
+
+  const handlePercentClick = (percent: number) => {
+    const cryptoAmount = bnOrZero(cryptoStakeBalanceHuman)
+      .times(percent)
+      .dp(asset.precision, BigNumber.ROUND_DOWN)
+    const fiatAmount = bnOrZero(cryptoAmount).times(marketData.price)
+    setValue(Field.FiatAmount, fiatAmount.toString(), {
+      shouldValidate: true,
+    })
+    setValue(Field.CryptoAmount, cryptoAmount.toString(), {
+      shouldValidate: true,
+    })
+  }
```

Comment:
callbacks?

Reply:
Also from the Foxy implementation, will add it in the same stacked PR

Reply:
https://github.com/shapeshift/web/pull/2325